pilot/pkg/proxy/envoy/testdata/external-service-route-rule.yaml.golden

```diff
@@ -0,0 +1,14 @@
+hosts:
+  - "*.google.com"
+http:
+  - route:
+    - destination:
+        name: "*.google.com"
+      weight: 50
+    - destination:
+        name: "*.google.com"
+        subset: foobar
```

Comment:
This does not seem right. Two destinations, one with subset and one without subsets. Well, I guess its technically possible, but I can't see why it would help in any way. May be just have distinct services in the destination, and if same service, ensure that they all have subsets.

Reply:
The proto does not say subset name is required. I was assuming that no subset name => route to the complete set.

Reply:
Yup, I think @ijsnellf's behavior is what we want.